Commit & Push задачи по разворачиванию конструктора BJS2-85451

vzeten opened the pull request:
Не знаю, нужно ли было перемещать main в пакет с задачей или нужно было создать новый, я просто переместил.

Commit:
Commit & Push задачи по разворачиванию конструктора BJS2-85451

MarselYsup commented on src/main/java/school/faang/bjs2_85451_turnAround/TurnAround.java:
```diff
@@ -0,0 +1,11 @@
+
+public class TurnAround {
+    public int[] reverse(int[] array) {
+        int[] arrayReverse = new int[array.length];
```
1) стоит проверять, что массив не равен null, а также в случае если массив пустой или состоит из одного элемента можно не выполнять лишнюю логику

MarselYsup replied:
2) Можно обойтись без создания дополнительного массива. Вместо этого можно инвертировать элементы массива "на месте", экономя память. Попробуй создать два указателя на первый и последний элемент, пробегаться по массиву пока указатели не встретятся и менять местами у указателей значения в массиве - так твой код будет занимать меньше памяти и менять элементы у уже имеющегося массива

MarselYsup commented on src/main/java/school/faang/bjs2_85451_turnAround/TurnAround.java:
```diff
@@ -0,0 +1,11 @@
+package school.faang.bjs2_85451_turnAround;
+
+public class TurnAround {
+    public int[] reverse(int[] array) {
```
метод лучше сделать static - так тебе не придется создавать отдельный объект для вызова метода reverse, да и в зависимости от объекта логика метода не меняется

MarselYsup reviewed (commented):
Насчет комментария выше, да, лучше всего создавать отдельные подпакеты для каждой задачи и в нем писать в одном классе логику задачи, в другом Main - демонстрацию кода либо тесты

MarselYsup commented on src/main/java/school/faang/bjs2_85451_turnAround/Main.java:
```diff
@@ -0,0 +1,20 @@
+        System.out.println("Исходный массив: " + Arrays.toString(arrayReal));
+        System.out.println("Перевернутый массив: " + Arrays.toString(arrayRealReverse));
+
+
```
много лишних пустых строк

Commit:
1. Добавил проверку чтобы массив не равнялся null и имел количество элементов больше 1.
2. Исправил логику метода - теперь элементы из начала массива перекладываются в конец массива и наоборот.
3. Сделал метод static чтобы более удобно вызывать его в классе Main
4. Убрал лишние пустые строки
